### ISSUE TITLE
Spreading props causes issues with unused-fields

### ISSUE THREAD

**Issue by VictorGlindasPaf** [relayjs/eslint-plugin-relay#116]
Passing fields into components by spreading shows them as unused by the `relay/unused-fields`, while passing them one by one does not cause any issues.

```ts
type Props = {
  data: ParentComponent_data;
};

const ParentComponent: FC<Props> = ({ data }) => {
  const { title, ...rest } = data;
  return <SharedComponent title={title} {...rest} />;
}

export default createFragmentContainer(ParentComponent, {
  data: graphql`
    fragment ParentComponent_data on Data {
       name # <- Triggers a warning about not being used!
       title
    }
  `,
});
```

**Comment by shogunsea** [comment]
I think this is essentially because this rule only does very naive static analysis of what properties are referenced in javascript file:
https://github.com/relayjs/eslint-plugin-relay/blob/fcacf1ed69097dd3a47856be8984de45a81c92f7/src/rule-unused-fields.js#L125-L129

it does **not** track how a piece of data are passed down to the child components and figure out if subfields are used in that child component.

Also there's a 2nd issue of this naive implementation, it can produce false positives because it's only checking the name of the properties, regardless of from which object it's being accessed, so the following is actually invalid case (`foo.bar` unsued)  but test still passes:

```
    graphql\`fragment foo on Foo {
      bar
      baz {
        bar
      }
    }\`;
    foo.baz.bar;
```
